container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/Node.java

```diff
@@ -21,19 +21,26 @@
     private final AtomicBoolean statusIsKnown = new AtomicBoolean(false);
     private final AtomicBoolean working = new AtomicBoolean(true);
     private final AtomicLong activeDocuments = new AtomicLong(0);
-    private final AtomicBoolean pendingPing = new AtomicBoolean();
+    private final AtomicLong pingSequence = new AtomicLong(0);
+    private final AtomicLong lastPing = new AtomicLong(0);
 
     public Node(int key, String hostname, int group) {
         this.key = key;
         this.hostname = hostname;
         this.group = group;
     }
 
-    /** Only send ping if this method return true. If not the is a ping outstanding. */
-    public boolean sendPing() { return ! pendingPing.getAndSet(true); }
-
-    /** Need to be called when a pong is called to allow next ping to go through. */
-    public void receivePing() { pendingPing.set(false); }
+    /** Give a monotonically increasing sequence number.*/
+    public long createPingSequenceId() { return pingSequence.incrementAndGet(); }
+    /** Checks if this pong is received in line and accepted, or out of band and should be ignored..*/
+    public boolean isLastReceivedPong(long pingId ) {
+        long last = lastPing.get();
+        while ((pingId > last) && ! lastPing.compareAndSet(last, pingId)) {
+            last = pingSequence.get();
```

Comment:
mixing sent and received ping sequence number here seems like a bad idea

Reply:
Fixed.